configs/sample_params/src/alias.rs

```diff
@@ -0,0 +1,93 @@
+//! Aliases that resolve with reference to [`crate::SampleParams`].
+
+use std::str::FromStr;
+
+use iroha_data_model::prelude::{AccountId, AssetId};
+
+/// Alias of `T`. Can be parsed to `T: FromStr` when the string is provided by [`resolve`](Alias::resolve).
+pub trait Alias<T: FromStr>
+where
+    <T as FromStr>::Err: core::fmt::Debug,
+{
+    /// Parse [`self`] to `T`, assuming [`self`] should successfully [`resolve`](Alias::resolve).
+    ///
+    /// # Panics
+    ///
+    /// - [`resolve`](Alias::resolve) implementation is not compatible with [`FromStr`] for `T`
+    ///
+    /// # Example
+    ///
+    /// See [`crate`] documentation.
+    #[inline]
+    fn parse_alias(&self) -> T {
+        self.resolve()
+            .as_ref()
+            .parse()
+            .expect("alias should resolve to compatible string with FromStr")
+    }
+    /// Resolve [`self`] to string that should be parsed to `T: FromStr`.
+    fn resolve(&self) -> impl AsRef<str>;
+}
+
+impl Alias<AccountId> for str {
+    fn resolve(&self) -> impl AsRef<str> {
+        let (name, domain) = self
+            .rsplit_once('@')
+            .expect("name@domain format should be given");
+        let sp = super::SampleParams::default();
```

Comment:
For the separate issue we can wrap `SampleParams` into smt like [`OnceLock`](https://doc.rust-lang.org/std/sync/struct.OnceLock) to initialize only once.